packages-exp/auth-exp/src/platform_browser/recaptcha/recaptcha_loader.ts

```diff
@@ -0,0 +1,139 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { querystring } from '@firebase/util';
+
+import { AUTH_ERROR_FACTORY, AuthErrorCode } from '../../core/errors';
+import { Delay } from '../../core/util/delay';
+import { Auth } from '../../model/auth';
+import { AuthWindow } from '../auth_window';
+import * as jsHelpers from '../load_js';
+import { Recaptcha } from './recaptcha';
+import { MockReCaptcha } from './recaptcha_mock';
+
+// ReCaptcha will load using the same callback, so the callback function needs
+// to be kept around
+export const _JSLOAD_CALLBACK = jsHelpers._generateCallbackName('rcb');
+const NETWORK_TIMEOUT_DELAY = new Delay(30000, 60000);
+const RECAPTCHA_BASE = 'https://www.google.com/recaptcha/api.js?';
+
+export interface ReCaptchaLoader {
+  load(auth: Auth, hl?: string): Promise<Recaptcha>;
+  clearedOneInstance(): void;
+}
+
+function pullGrecaptcha(): Recaptcha | undefined {
+  return (window as AuthWindow).grecaptcha;
+}
+
+const WINDOW = window as AuthWindow;
+
+/**
+ * Loader for the GReCaptcha library. There should only ever be one of this.
+ */
+export class ReCaptchaLoaderImpl implements ReCaptchaLoader {
+  private hl = '';
+  private counter = 0;
+  private readonly librarySeparatelyLoaded = !!pullGrecaptcha();
```

Comment:
why does this need to a be a method? can't we just call WINDOW.grecaptcha?

Reply:
Done